app/code/Magento/ConfigurableProduct/Model/Product/Type/Configurable.php

```diff
@@ -1294,9 +1292,11 @@ private function loadUsedProducts(\Magento\Catalog\Model\Product $product, $cach
             $usedProducts = $this->readUsedProductsCacheData($cacheKey);
             if ($usedProducts === null) {
                 $collection = $this->getConfiguredUsedProductCollection($product);
-                if ($salableOnly) {
-                    $collection = $this->salableProcessor->process($collection);
-                }
+                $collection->setFlag('has_stock_status_filter',false);
```

Comment:
Please remove this line: `$collection->setFlag('has_stock_status_filter',false);`